test/integration/api-schema/basic.js

```diff
@@ -24,50 +24,74 @@ const next = afterAll(function (err, validators) {
   test('metadata schema failure', function (t) {
     t.strictEqual(validateMetadata({}), false)
     validateFieldMessages(t, validateMetadata.errors, [
-      { field: 'data.service', message: 'is required' }
+      { message: 'should have required property \'service\'' }
     ])
     t.end()
   })
 
   test('transaction schema failure', function (t) {
     t.strictEqual(validateTransaction({}), false)
     validateFieldMessages(t, validateTransaction.errors, [
-      { field: 'data.duration', message: 'is required' },
-      { field: 'data.type', message: 'is required' },
-      { field: 'data.id', message: 'is required' },
-      { field: 'data.trace_id', message: 'is required' },
-      { field: 'data.span_count', message: 'is required' }
+      { message: 'should have required property \'duration\'' },
+      { message: 'should have required property \'id\'' },
+      { message: 'should have required property \'span_count\'' },
+      { message: 'should have required property \'trace_id\'' },
+      { message: 'should have required property \'type\'' }
     ])
     t.end()
   })
 
   test('span schema failure', function (t) {
     t.strictEqual(validateSpan({}), false)
     validateFieldMessages(t, validateSpan.errors, [
-      { field: 'data.duration', message: 'is required' },
-      { field: 'data.name', message: 'is required' },
-      { field: 'data.type', message: 'is required' },
-      { field: 'data.id', message: 'is required' },
-      { field: 'data.trace_id', message: 'is required' },
-      { field: 'data.parent_id', message: 'is required' },
-      { field: 'data', message: 'no schemas match' }
+      { message: 'should have required property \'duration\'' },
+      { message: 'should have required property \'id\'' },
+      { message: 'should have required property \'name\'' },
+      { message: 'should have required property \'parent_id\'' },
+      { message: 'should have required property \'trace_id\'' },
+      { message: 'should have required property \'type\'' },
+      { message: 'should have required property \'start\'' },
+      { message: 'should have required property \'timestamp\'' },
+      { message: 'should match some schema in anyOf' }
     ])
```

Comment:
The `start`, `timestamp`, and `should match some schema in anyOf` messages don't have counterparts in the old version of the code.  Presuming these are necessary to bring us in line with the new schema docs and/or new validator.

Reply:
Yah. Perhaps subtle in this is that I've elided some of the fields returned by the validator. For this example the validator returns:

```
[ { keyword: 'required',
    dataPath: '',
    schemaPath: '#/required',
    params: { missingProperty: 'duration' },
    message: 'should have required property \'duration\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/required',
    params: { missingProperty: 'id' },
    message: 'should have required property \'id\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/required',
    params: { missingProperty: 'name' },
    message: 'should have required property \'name\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/required',
    params: { missingProperty: 'parent_id' },
    message: 'should have required property \'parent_id\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/required',
    params: { missingProperty: 'trace_id' },
    message: 'should have required property \'trace_id\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/required',
    params: { missingProperty: 'type' },
    message: 'should have required property \'type\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/anyOf/0/required',
    params: { missingProperty: 'start' },
    message: 'should have required property \'start\'' },
  { keyword: 'required',
    dataPath: '',
    schemaPath: '#/anyOf/1/required',
    params: { missingProperty: 'timestamp' },
    message: 'should have required property \'timestamp\'' },
  { keyword: 'anyOf',
    dataPath: '',
    schemaPath: '#/anyOf',
    params: {},
    message: 'should match some schema in anyOf' } ]
```

Those three new entries are a better representation of the "anyOf" block (https://github.com/elastic/apm-server/blob/89e4f81d352fef73808ecf5cdba5db2c112533b0/docs/spec/v2/span.json#L697-L718) for which the "is-my-json-valid" validator returned:

```
{ field: 'data', message: 'no schemas match' }
```